[Failing test] gce-cos-<branch>-ingress are timing out

**Which jobs are failing**:
* [`ci-kubernetes-e2e-gci-gce-ingress`](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-ingress/20602)
* [`ci-kubernetes-e2e-gce-cos-k8sbeta-ingress`](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-cos-k8sbeta-ingress/6339)

**Which test(s) are failing**:
The jobs are timing out, but just before that, the following tests failed:
* `[sig-network] Loadbalancing: L7 GCE [Slow] [Feature:Ingress] should be able to switch between HTTPS and HTTP2 modes`
* `[sig-network] Loadbalancing: L7 GCE [Slow] [Feature:Ingress] should update ingress while sync failures occur on other ingresses`
* `[sig-network] Loadbalancing: L7 GCE [Slow] [Feature:NEG] should create NEGs for all ports with the Ingress annotation, and NEGs for the standalone annotation otherwise` (in `gce-cos-master-ingress`)
* `[sig-network] Loadbalancing: L7 GCE [Slow] [Feature:Ingress] should be able to switch between HTTPS and HTTP2 modes` (in `gce-cos-1.14-ingress`)

**Since when has it been failing**:
late 2019-03-07

**Testgrid link**:
* https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-ingress
* https://testgrid.k8s.io/sig-release-1.14-blocking#gce-cos-1.14-ingress

**Reason for failure**:
Timeout related to the L7 controller:
```
error: L7 controller failed to delete all cloud resources on time. timed out waiting for the condition
```

/sig network
/priority critical-urgent (failing consistently in release blocking jobs, blocking other tests entirely from running)

cc @kubernetes/sig-network-test-failures 
cc @smourapina @alejandrox1 @kacole2 @mortent

Seems like Ingress controller kept panicing during the test (https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-cos-k8sbeta-ingress/6367/artifacts/test-6724261826-master/glbc.log):
```
I0311 14:12:41.510138       1 backends.go:249] Sync: backends [{ingress-5906/echoheaders/443 32165 443 HTTPS 8443 false <nil>}]
E0311 14:12:42.100687       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/k8s.io/ingress-gce/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/k8s.io/ingress-gce/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/k8s.io/ingress-gce/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:388
/go/src/k8s.io/ingress-gce/pkg/healthchecks/healthchecks.go:332
/go/src/k8s.io/ingress-gce/pkg/healthchecks/healthchecks.go:244
/go/src/k8s.io/ingress-gce/pkg/healthchecks/healthchecks.go:106
/go/src/k8s.io/ingress-gce/pkg/backends/backends.go:220
/go/src/k8s.io/ingress-gce/pkg/backends/backends.go:283
/go/src/k8s.io/ingress-gce/pkg/backends/backends.go:252
/go/src/k8s.io/ingress-gce/pkg/controller/cluster_manager.go:98
/go/src/k8s.io/ingress-gce/pkg/controller/controller.go:349
/go/src/k8s.io/ingress-gce/pkg/controller/controller.go:299
/go/src/k8s.io/ingress-gce/pkg/controller/controller.go:111
/go/src/k8s.io/ingress-gce/pkg/utils/taskqueue.go:84
/go/src/k8s.io/ingress-gce/pkg/utils/taskqueue.go:54
/go/src/k8s.io/ingress-gce/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/k8s.io/ingress-gce/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/k8s.io/ingress-gce/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/go/src/k8s.io/ingress-gce/pkg/utils/taskqueue.go:54
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1674e63]
```

/cc @rramkumar1

--- a/test/e2e/network/ingress.go
+++ b/test/e2e/network/ingress.go
@@ -427,7 +427,8 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
 			// TODO(nikhiljindal): Check the instance group annotation value and verify with a multizone cluster.
 		})
 
-		It("should be able to switch between HTTPS and HTTP2 modes", func() {
+		// TODO: remove [Unreleased] tag to once the new GCE API GO client gets revendored in ingress-gce repo
+		It("should be able to switch between HTTPS and HTTP2 modes [Unreleased]", func() {
 			httpsScheme := "request_scheme=https"
 
 			By("Create a basic HTTP2 ingress")

